src/main.rs

```diff
@@ -9,69 +9,168 @@ mod utils;
 #[cfg(test)]
 pub mod tests;
 
-use crate::compiler::AmberCompiler;
-use clap::Parser;
+use crate::compiler::{AmberCompiler, CompilerOptions};
+use clap::{Args, CommandFactory, Parser, Subcommand};
+use clap_complete::Shell;
 use colored::Colorize;
 use heraclitus_compiler::prelude::*;
 use std::error::Error;
-use std::fs;
 use std::io::{prelude::*, stdin};
-use std::path::{Path, PathBuf};
+use std::path::PathBuf;
 use std::process::Command;
+use std::{fs, io};
 
 #[derive(Parser, Clone, Debug)]
 #[command(version, arg_required_else_help(true))]
-#[derive(Default)]
-pub struct Cli {
-    #[arg(help = "'-' to read from stdin")]
+struct Cli {
+    #[command(subcommand)]
+    command: Option<CommandKind>,
+
+    /// Input filename ('-' to read from stdin)
     input: Option<PathBuf>,
-    #[arg(help = "'-' to output to stdout, '--silent' to discard")]
-    output: Option<PathBuf>,
 
+    /// Arguments passed to Amber script
+    #[arg(trailing_var_arg = true)]
+    args: Vec<String>,
+}
+
+#[derive(Subcommand, Clone, Debug)]
+enum CommandKind {
+    /// Execute Amber code fragment
+    Eval(EvalCommand),
+    /// Execute Amber script
+    Run(RunCommand),
+    /// Check Amber script for errors
+    Check(CheckCommand),
+    /// Compile Amber script to Bash
+    Build(BuildCommand),
+    /// Generate Amber script documentation
+    Docs(DocsCommand),
+    /// Generate Bash completion script
+    Comp,
```

Comment:
Can we rename `Comp` so that it's not too similar to a word _Compile_. This could confuse some users. How about `Autocomplete`, `Autocomp`, `Compl`, `Completion`...

Reply:
Renamed as `Completion`.